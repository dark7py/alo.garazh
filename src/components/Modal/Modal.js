import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import ClickOutside from 'components/ClickOutside';
import './Modal.scss';

class Modal extends PureComponent {
  // Отключаем скрол пока открыто модальное окно
  body = document.getElementsByTagName('body')[0];

  componentDidMount() {
    this.body.style.overflow = 'hidden';
  }

  componentWillUnmount() {
    this.body.style.overflow = 'auto';
  }

  render() {
    const { closeModal, children } = this.props;

    return (
      <div className="modal__wrapper">
        <ClickOutside onClickOutside={closeModal}>
          <div className="modal__container">
            <button
              type="button"
              className="modal__close"
              onClick={closeModal}
            />
            {children}
          </div>
        </ClickOutside>
      </div>
    );
  }
}

Modal.propTypes = {
  closeModal: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired
};

export default Modal;
