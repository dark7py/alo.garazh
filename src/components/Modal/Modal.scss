.modal__wrapper {
  position: fixed;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  background-color: rgba(32, 32, 32, 0.1);
  z-index: 100;
}

.modal__container {
  display: flex;
  flex-direction: column;

  position: absolute;
  top: 50%;
  left: 50%;
  max-width: 720px;
  min-width: 320px;
  border-radius: 16px;
  padding: 50px 100px;
  text-align: center;
  transform: translate(-50%, -50%);
  background: white;
  box-shadow: 0 8px 24px 0 rgba(32, 32, 32, 0.1);

  @media (max-width: 650px) {
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    transform: none;

    padding: 50px 25px;
    border-radius: 0;
    overflow: scroll;
  }
}

.modal__close {
  position: absolute;
  top: 30px;
  right: 30px;

  display: block;
  width: 24px;
  height: 24px;
  padding: 0;
  border: none;
  margin: 0;

  background-image: url(/images/icon-close.svg);
  background-repeat: no-repeat;
  background-position: center;
  background-color: transparent;

  cursor: pointer;

  @media (max-width: 650px) {
    top: 10px;
    right: 10px;
  }
}
