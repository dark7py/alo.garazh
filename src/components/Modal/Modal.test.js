import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Modal from './Modal';

test('Modal does not crash', () => {
  const mockFunction = jest.fn();

  const tree = shallow(
    <Modal closeModal={mockFunction}>
      content
    </Modal>
  );

  expect(toJson(tree)).toMatchSnapshot();
});
