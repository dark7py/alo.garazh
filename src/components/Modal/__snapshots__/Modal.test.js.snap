// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal does not crash 1`] = `
<div
  className="modal__wrapper"
>
  <ClickOutside
    onClickOutside={[MockFunction]}
  >
    <div
      className="modal__container"
    >
      <button
        className="modal__close"
        onClick={[MockFunction]}
        type="button"
      />
      content
    </div>
  </ClickOutside>
</div>
`;
